cloudinit/cmd/main.py

```diff
@@ -143,7 +144,7 @@ def parse_cmdline_url(cmdline, names=("cloud-config-url", "url")):
     raise KeyError("No keys (%s) found in string '%s'" % (cmdline, names))
```

Comment:
Please add some doc/rtd/topics/instancedata.rst content 


+ * `#cloud-config` provided by one of the kernel command-line  parameters `cloud-config-url` or a yaml content beginning with `cc:` and URL-encoded YAML content ending with `end_cc`.


I might be nice to define this supported behavior of `cloud-config-url` (expecting `#cloud-config` header as the first line) I didn't see any current docs that describe this.

Reply:
I have rescued the lost `doc/sources/kernel-cmdline.txt`, which was not rendered, converted it to rst and referenced it from `instancedata.rst` and `format.rst`.

> or a yaml content beginning with cc: and URL-encoded YAML content ending with end_cc

I am not sure about what to do about this.

Reply:
Let's hold on this additional documentation for a separate PR and we can discuss. I think it belongs in kernel-cmdline page, but we can sort it tomorrow.